font

```diff
@@ -158,3 +158,40 @@ sub on_user_command
     $self->change_size(($1 eq "increment") ? +1 : -1);
   }
 }
+
+sub on_init
+{
+  my ($self) = @_;
+
+  ($self->{winx}, $self->{winy}) = (0, 0);
+}
+
+sub get_dpi_at
+{
+  my ($px, $py) = @_;
+
+  foreach (grep {m/ connected /} `xrandr`) {
```

Comment:
This adds a dependency, which I guess I'm okay with. I'd say `xrandr` is pretty much essential on gui linux nowadays.  But both it and `xdpyinfo` do not always produce correct output.  There is also `xwininfo`.

Can you think of a way to make executing the branch of the code that checks/ scales DPI optional or based on a setting?  Alternatively, or in addition, what if xrandr is not found?